sno/log.py

```diff
@@ -0,0 +1,75 @@
+from datetime import datetime, timezone, timedelta
+import subprocess
+import sys
+
+import click
+
+from .exec import execvp
+from .exceptions import SubprocessError
+from .output_util import dump_json_output
+from .timestamps import datetime_to_iso8601_utc, timedelta_to_iso8601_tz
+
+
+@click.command(context_settings=dict(ignore_unknown_options=True,))
+@click.pass_context
+@click.option(
+    "--output-format",
+    "-o",
+    type=click.Choice(["text", "json"]),
+    default="text",
+    help=(
+        "Output format. 'quiet' disables all output and implies --exit-code.\n"
+        "'html' attempts to open a browser unless writing to stdout ( --output=- )"
+    ),
+)
+@click.option(
+    "--json-style",
+    type=click.Choice(["extracompact", "compact", "pretty"]),
+    default="pretty",
+    help="How to format the output. Only used with --output-format=json",
+)
+@click.argument("args", nargs=-1, type=click.UNPROCESSED)
+def log(ctx, output_format, json_style, args):
+    """ Show commit logs """
+    if output_format == "text":
+        execvp("git", ["git", "-C", ctx.obj.repo.path, "log"] + list(args))
+
+    elif output_format == "json":
+        repo = ctx.obj.repo
+        try:
+            cmd = ["git", "-C", repo.path, "log", "--pretty=format:%H"] + list(args)
+            r = subprocess.run(cmd, encoding="utf8", check=True, capture_output=True)
```

Comment:
Ideally we'd iterate over the output of this while it's still running and dump json output in a generative way rather than capturing all the output before dumping any of it. I think we probably have all the pieces (`json.dump` apparently starts producing output right away.) But doing that would add a lot of complexity, and probably we'll need some really long-lived repos for the size of these logs to become a problem. Maybe add a TODO for that?